### ISSUE TITLE
Diffusion model generating identical images after DDPO fine-tuning 

### ISSUE THREAD

**Issue by shashankg7** [huggingface/trl#2113]
### System Info

- `transformers` version: 4.44.2
- Platform: Linux-5.15.0-1068-aws-x86_64-with-glibc2.31
- Python version: 3.9.19
- Huggingface_hub version: 0.24.7
- Safetensors version: 0.4.5
- Accelerate version: 0.34.2
- Accelerate config:    not found
- PyTorch version (GPU?): 2.4.1+cu121 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: Yes, trying in a multiGPU node

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [X] My own task or dataset (give details below)

### Reproduction

I fine-tuned a SD model using the DDPO model following the script https://github.com/huggingface/trl/blob/main/examples/scripts/ddpo.py. I have a custom reward model and a prompt dataset. I fine-tuned on a small prompt set (30 prompts) to check if the model is overfitting on the rewards. The rewards are increasing over steps. 

For inference I used the following script to generate images for some sample prompts:

```
import torch
from trl import DefaultDDPOStableDiffusionPipeline

torch.manual_seed(42)
### Pre-trained model inference
pipeline = DefaultDDPOStableDiffusionPipeline(
        "runwayml/stable-diffusion-v1-5"
    )

device = torch.device("cuda")
pipeline.vae.to(device)
pipeline.text_encoder.to(device)
pipeline.unet.to(device)


prompts = ["Tom Cruise in a yellow suit and a pink tie", 
           "A red elephant next to a brown dog", 
           "A blue building next to a red elephant"]
results = pipeline(prompts)

for prompt, image in zip(prompts, results.images):
    image.save(f"./example_images/{prompt}_pretrained.png")

#### Fine-tuned model inference

prompts = ["Tom Cruise in a yellow suit and a pink tie", 
           "A red elephant next to a brown dog", 
           "A blue building next to a red elephant"]



pipeline = DefaultDDPOStableDiffusionPipeline(
        "runwayml/stable-diffusion-v1-5"
    )
#pipeline.sd_pipeline.load_lora_weights("/checkpoints/shashankgupta/ddpo_test2")

device = torch.device("cuda")

pipeline.vae.to(device)
pipeline.text_encoder.to(device)
pipeline.unet.to(device)

results = pipeline(prompts)

for prompt, image in zip(prompts, results.images):
    image.save(f"./example_images/{prompt}_ft.png")
```

Model was generating different images until I realised that I didn't use manual_seed when loading the fine-tuned model. When I apply manual_seed the second time, the fine-tuned model generates almost identical images. 

### Expected behavior

After RL fine-tuning, the diffusion model should generate different images for the same prompt.

**Comment by shashankg7** [comment]
Update: I trained with LoRA fine-tuning, which was resulting in similar models. Fine-tuned the entire model, it generates different images.

**Comment by shashankg7** [comment]
Update: I trained with LoRA fine-tuning, which was resulting in similar models. Fine-tuned the entire model, it generates different images.
